activerecord/lib/arel/visitors/to_sql.rb

```diff
@@ -913,6 +922,14 @@ def has_join_sources?(o)
           o.relation.is_a?(Nodes::JoinSource) && !o.relation.right.empty?
         end
 
+        def has_table_alias(o)
+          o&.relation&.instance_values.respond_to?(:table_alias) && !o.relation.table_alias.nil?
+        end
+
+        def has_mysql_adapter(connection)
```

Comment:
This is good to know! I implemented this method in the `visitors/mysql.rb` file so that it would override the method in `visitors/to_sql.rb`.  I hope that's an acceptable solution, let me know if not.